### ISSUE TITLE
AH: Filter Custom Data Download By Owner/ Operator 

### ISSUE THREAD

**Issue by JanellC** [US-EPA-CAMD/PI-1#1107]
### Definition of Ready
- [x] Does this meet the INVEST criteria?
- [x] Does this have a clearly defined user?
- [x] Does this have an estimate?
- [x] Does this have acceptance criteria?
- [x] Does this have sufficient context (wireframes, business rules, etc.)?
- [x] Is this traceable to a feature, epic, or PI?

Dependency: Add endpoint to Facilities API to return list of Owner/Operators
## User Story
As a CAMPD custom data download user I want to enter filter criteria for owner/operator so that my dataset includes data for my desired owners/ operators 

## Acceptance Criteria
Given that a user has selected a data type and sub type
When the filter criteria page is displayed
Then user has the option to select owner/ operator 

Given that a user has selected owner/ operator 
When the functionality is displayed
Then the results populate in a list

Given that a user has selected owner/ operator 
When the functionality is displayed
Then the user can select one or multiple owners/ operators 

Given that a user has selected owner/ operator 
When the search is executed
Then the system conducts a “contains” search 

Given that a user has selected owner/ operator 
When the search is executed and no results are found
Then the system notifies the user that "No owners/ operators match your search"

Given that a user has applied filter criteria for owner/ operator 
When the criteria is applied
Then the system displays a filter tag

Given that a user has applied filter criteria for owner/ operator 
When any owner/operator criteria is cleared
Then the system clears the criteria from the filter

Given that a user has selected preview data 
When the functionality is displayed 
Then the data preview includes filter criteria for the selected owner/ operator(s)

### Does this Story require 508 Testing?
- [ ] Implementing a new authoring tool that will produce code / content for the application
- [ ] Implementing a 3rd party widget that will be part of the application
- [x] Adding new / updating custom functionality 
- [ ] Adding new / updating a page template
- [ ] Adding new / replacing existing functionality (unless the replacement has already passed 508 testing)
- [ ] Full Site Review prior to a scheduled EPA 508 review
- [ ] Planning a video that will appear on the application
- [ ] Producing a video that will appear on the application
- [ ] Producing Training materials
- [ ] Producing documents

## Context
When there are owners and operators with the same name (e.g. Alabama Power company), then the filter will display the name only once. 

**Filter Tags** 
Populates filter tag on Apply Filter
if 1 parameter is selected, then show parameter in tag
Tag will remain fixed width, if parameters text exceeds fixed width, then full name is displayed in hover text/Tool tip
If more than 1, show number of parameters
Clicking x in tag clear filter
Click on the tag opens filter in panel
Will provide the option to clear all filters in tag group
Tags will have a hover functionality to display the list of selections

**Functionality** 

Owner
Dynamic search
Search is a contains search on all allowance accounts 
Results populate in a list, parameters populates when a value is selected from the list (results list should stay at top of page/visible in current view, possible use pagination to keep window small)
Search is not case sensitive
If no results are found, the system will indicate so. “No accounts match your search.”
Multi select
Owner/operators selected in the list will have visual indicator that they been selected (check mark/ greyed out)

Operator
Dynamic search
Search is a contains search on  on all allowance accounts 
Results populate in a list, parameters populates when a value is selected from the list (results list should stay at top of page/visible in current view, possible use pagination to keep window small)
Search is not case sensitive
If no results are found, the system will indicate so. “No accounts match your search.”
Multi select
Owner/operators selected in the list will have visual indicator that they been selected (check mark/ greyed out)

**Data Preview**

Scope includes data preivew for All allowance holdings holdings parameters 

Refer to the CAMPD_ DataLabels_Allowance file [here ](https://teams.microsoft.com/l/file/C57F5045-F938-472B-AE6E-51BF711FFA12?tenantId=88b378b3-6748-4867-acf9-76aacbeca6a7&fileType=xlsx&objectUrl=https%3A%2F%2Fusepa.sharepoint.com%2Fsites%2FCAMDCVPTeam%2FShared%20Documents%2FGeneral%2FCoding%20%26%20Error%20Standards%2FCAMPD_DataLables_Allowance_5_12_2021.xlsx&baseUrl=https%3A%2F%2Fusepa.sharepoint.com%2Fsites%2FCAMDCVPTeam&serviceName=teams&threadId=19:6f05dfa026c84672a2cc1bfa72c58f14@thread.skype&groupId=21a5e1cd-d3f1-48a6-801f-0d34e2d63d23)

For each record that matches the criteria, the system will display in a tabular format:

ACCOUNT_NUMBER
ACCOUNT_NAME
ORISPL_CODE
PRG_CODE
VINTAGE_YEAR
TOTAL_BLOCK
START_BLOCK
END_BLOCK
STATE
EPA_REGION (2 digit number)
OWN_DISPLAY
ACCOUNT_TYPE

Results will display total record count (Viewing the first 100 of X records)
Results are limited to 100 records (Allowance Holdings API  is ordered by PRG_CODE > ACCOUNT_NUMER > VINTAGE_YEAR > START_BLOCK )
Results in preview are limited to the first 100 records
Results in preview will have a scroll functionality to show all columns/rows
Results columns are are sortable in ascending and descending order (alphanumeric order) . Sort is limited to the original 100 results

## Testing
Test Case: [TC342](https://cvpcorp.testquality.com/project/11428/plan/21361/test/299305) 
Tester: @JanellC 

508 Testing: 
Tester: @aprematta 

### Definition of Done
- [x] Have all Acceptance Criteria been agreed to and validated by the scrum team?
- [x] Has the code been unit tested and peer reviewed?
- [x] Have the functional tests been executed?
- [x] Have all defects been reviewed and dispositioned, resolved, or deferred?
- [ ] Has the user story been accepted by the Product Owner?
- [ ] Is the user story ready to be deployed to the test/staging environments?

**Comment by JorjaComer** [comment]
@JanellC see updated context based on a decision made in DSU on 9/3.

**Comment by JanellC** [comment]
TC342 failed @tlageman-epa. I logged Bug: Filter Custom Data Download By Owner/ Operator #1991 to track the changes

**Comment by JanellC** [comment]
Bug: Filter Custom Data Download By Owner/ Operator #1991 addressed. This ticket is ready for 508 testing @aprematta
